src/object/obj_class.c

```diff
@@ -370,9 +370,33 @@ daos_oclass_id2name(daos_oclass_id_t oc_id, char *str)
 	}
 
 	D_ASSERT(oc->oc_id == OC_UNKNOWN);
+	strcpy(str, "UNKNOWN");
 	return -1;
 }
 
+/** Return the list of registered oclass names */
+size_t
+daos_oclass_names_list(size_t size, char *str)
+{
+	struct daos_obj_class   *oc;
+	size_t len = 0;
+
+	if (size <= 0 || str == NULL)
+		return -1;
+
+	*str = '\0';
+	for (oc = &daos_obj_classes[0]; oc->oc_id != OC_UNKNOWN; oc++) {
+		if (oc->oc_name == NULL)
+			break;
+		len += strlen(oc->oc_name) + 2;
+		if (len < size) {
+			strcat(str, oc->oc_name);
+			strcat(str, ", ");
```

Comment:
very minor - you may choose to ignore. this leaves a comma at the end of the list of object class names in the output. maybe do special case for first class name (just print the name) and then for all others print comma then the name?